### DISCUSSION TITLE
External Selector Does Not Trigger State Change

### DISCUSSION THREAD

**Discussion by elson-currentcorp** [pmndrs/zustand#2536] [Bug report]
I have a modal that uses the states `value`, `setValue`, `isOpen` and `setIsOpen` located in the store. Whenever this modal is opened, I want to reset the input field to be whatever the current `value` in the store is. The user can edit the input field, this does not affect the stored `value` until the user clicks save.

This is my basic implementation:
https://codesandbox.io/p/sandbox/amazing-rhodes-forked-ydqmgd?file=%2Fsrc%2FApp.tsx%3A46%2C1

**Why does line 42 work, but not line 45?** The modal open state does not change if I use a external selector even though the state changes.

**Comment by dai-shi** [comment]
Okay, this is interesting. I don't think you are doing anything wrong.
Usually, stable selectors are good for optimization.
Calling setState in render is also good as react docs suggest, but it's a new pattern.

Zustand v5-alpha actually works fine. It seems like a bug in useSyncExternalStoreWithSelector.
I wonder if anyone is interested in creating a reproduction without zustand and report it to react repo.
